scripts/vehicle_node.py

```diff
@@ -0,0 +1,380 @@
+
+        #Create an list and fill the list with x, y values and z the values will be the filtered gaussian noise
+        incr_x = 0
+        incr_y = 0
+        cloud = []
+        for x in np.arange(-height, (height+1), resolution):
+            for y in np.arange(-width, (width+1), resolution):
+                innerlist = []
+                innerlist.append(x)
+                innerlist.append(y)
+                innerlist.append(gaussian_array[incr_x][incr_y])
+                cloud.append(innerlist)
+                incr_y+=1
+            incr_x+= 1
+            incr_y = 0
+
+        return cloud
+
+
+    def velocity_cmd_callback(self, data):
+
+        #assign updated command velocities to twist variable
+        self.twist = data
+
+
+    def update_position(self, event):
+
+        resolution =  rospy.get_param('/resolution')
+        #time elapsed since last update position call (using event.current_real - event.last_real throws an error on first run as event.last_real does not exist in the first run)
+        time = 0.05
+
+        #Calculate angle turned in the given time using omega = theta/time
+        angle = self.twist.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = self.twist.linear.x*time
+
+        #Calculate yaw, pitch and roll of the robot (pitch and roll currently not calculated)
+        self.pose.orientation.x = 0.0
+        self.pose.orientation.y = 0.0
+        self.pose.orientation.z += angle
+
+        #Calculate vehicle x, y, z position coordinates 
+        self.pose.position.x += (distance)*cos(self.pose.orientation.z)
+        self.pose.position.y += (distance)*sin(self.pose.orientation.z)
+
+        myarray = np.asarray(self.cloud)
+
+        points = []
+        values = []
+
+        for x in range(len(myarray)):
+            if (self.pose.position.x - 0.05 - resolution) <= myarray[x,0] <=  (self.pose.position.x + 0.05 + resolution):
+                if (self.pose.position.y - 0.05 - resolution) <= myarray[x,1] <= (self.pose.position.y + 0.05 + resolution):
+                    innerlist = []
+                    innerlist.append(myarray[x,0])
+                    innerlist.append(myarray[x,1])
+                    values.append(myarray[x,2])
+                    points.append(innerlist)
+
+        x1, y1 = np.meshgrid(np.linspace((self.pose.position.x-0.05), (self.pose.position.x+0.05), 50), np.linspace((self.pose.position.y-0.05), (self.pose.position.y+0.05), 50))
+                
+        grid_z2 = griddata(points, values, (x1, y1), method='linear')
+
+        flat_list = []
+
+        for sublist in grid_z2:
+            for item in sublist:
+                flat_list.append(item)
+                
+        avg = sum(flat_list)/len(flat_list)
+
+        self.pose.position.z = avg
+
+
+    def publish_pose_msg(self, event):
+        
+        #Header
+        self.pose_stamped.header.stamp = rospy.Time.now()
```

Comment:
Nit: you can use event.current_real instead of the rospy.Time.now() to be a little more accurate. 
http://wiki.ros.org/rospy/Overview/Time